### DISCUSSION TITLE
error in await useFetch is always null on page reload

### DISCUSSION THREAD

**Discussion by ahoiroman** [nuxt/nuxt#17464] [Questions]
Hello there,

I got something I don't understand.

I created a page, which should immediately load data from the API. This is how I do that:

```js
<template>
    <nuxt-layout name="blank">
        //...
    </nuxt-layout>
</template>
<script setup>

const {data: verification, error } = await useFetch(`/proxy/api/does-not-exist`, {
    server: false,
    credentials: "include",
});

if (error.value) {
    //empty on page reload (F5), even if the ajax-call returns 404, 401, 403 or 422, but not empty, if I navigate through Nuxt-link to this page
} else {
    // ...
}
</script>
```

Now the strange thing is: This works fine, if I go to that page using a `nuxt-link`. The `error` is reported correctly and I can handle it in the if-statement.

But if I directly go to the page by pasting a link to the browsers address bar or if I press F5, the response is **still** 404, but the `error.value` is empty.

I don't understand that. Can someone help me out?

**Comment by ahoiroman** [comment]
I just mentioned, that if I wrap the useFetch in a setTimeout, the problem is gone...

```
setTimeout(async () => {
    const {data, error, pending } = await useFetch(`/proxy/api/does-not-exist`, {
        server: false,
        credentials: "include",
    });

    console.log(data,error,pending);

    console.log(error.value) //is always showing the correct error value
});
```

**Reply by ahoiroman** [reply]
I also mentioned that this is only happening, if `server: false` is set.

**Comment by danielroe** [comment]
This is deliberate and required. We cannot await on fetching on the initial client-side load with `server: false`, or it will cause a hydration error (different state b/t server + client).

**Reply by ahoiroman** [reply]
Okay. I understand.

Is there any documentation or sandbox you can recommend to me in order to learn how to handle this in my case?

Unfortunately I can not use `server: true`, as I am using Nuxt-auth with Laravel Sanctum and I cannot get the user's session cookie  to the server-side request via proxy resulting in HTTP 401.

**Reply by Diizzayy** [reply]
> Unfortunately I can not use server: true, as I am using Nuxt-auth with Laravel Sanctum and I cannot get the user's session cookie to the server-side request via proxy resulting in HTTP 401

@ahoiroman You should be able to utilize the [`useRequestHeaders` composable](https://v3.nuxtjs.org/api/composables/use-request-headers/) to proxy the user's cookies in SSR.

I think [the answer in this discussion](https://github.com/nuxt/framework/discussions/4504#discussioncomment-2619249) should lead you in the right direction

**Reply by ahoiroman** [reply]
Thanks, @Diizzayy 

I saw this post before, but the problem is that I can not get the session-cookie, as it's returning `null` - I guess it's because it has the httpOnly-flag.

I really appreciate your help - I will try to dig deeper :-)

**Reply by Diizzayy** [reply]
> I guess it's because it has the httpOnly-flag

@ahoiroman I can assure you that the `httpOnly` flag isn't the issue here, `useRequestHeaders` is a SSR utility and is indeed able to extract it.

If you're up for it, I'd be willing to have a look at your implementation and help you resolve this. You can reach out to me on discord - Diizzayy#1964
